src/main/scala/org/broadinstitute/hail/io/IndexedBinaryBlockReader.scala

```diff
@@ -44,10 +38,10 @@ class VariantRecord[K] extends Serializable {
   }
 }
 
-abstract class IndexedBinaryBlockReader[K](job: Configuration, split: FileSplit)
-  extends RecordReader[LongWritable, VariantRecord[K]] {
+abstract class IndexedBinaryBlockReader[T <: BinaryDecoder[_, _]](job: Configuration, split: FileSplit)
```

Comment:
Is this type bound necessary?  You don't seem to use the fact `T` is a `BinaryDecoder`.